reports/GitalyFail/GitalyFail.tla

```diff
@@ -0,0 +1,137 @@
+                /\ rmState[r] == "worknig"
+                /\ (\/ DoNothing
+                    \/ RMDoPrepare(r) )
+                                      
+RMDoPrepare(r) ==  /\ rmState' = [rmState EXCEPT ![r] = "prepared"]
+\*                /\ msgs' = msgs \union {[type |-> "Prepared", rm |-> r]}
+                   /\ RMConfirmPrepare(r)
+                   /\ UNCHANGED << tmState, tmPrepared >>
+
+RMConfirmPrepare(r) == \/ DoNothing
+                       \/ TMConfirmPrepare(r) 
+
+TMConfirmPrepare(r) == /\ rmPrepared' = rmPrepared \union r
+                       /\ msgs' = msgs \union {[type |-> "Prepared", rm |-> r]}
+        
+IsQuorumPrepared == /\ [type |-> "Abort"] \notin msgs
+                    /\ Cardinality(msgs) * 2 > Cardinality(Nodes)
+                    \* AND MasterNode voted for Commit                     
+
+TMSendCommit == /\ IsQuorumPrepared
+                /\ tmState = "init"
+                /\ tmState' = "done"
+                \* /\ tmPrepared = RM
+                /\ msgs' =  msgs \union {[type |-> "Commit"]}
+                /\ tmGeneration' = tmGeneration + 1
+                /\ UNCHANGED << rmState, tmPrepared >>  
+
+TMSendAbort ==  /\ tmState = "init"
+                /\ tmState' = "done"
+                /\ msgs' =  msgs \union {[type |-> "Abort"]}
+                /\ UNCHANGED << rmState, tmPrepared >>  
+
+\* Here it is possible to devide to DoNothing and DoAbort, but it does not matter for the inconsistency issue. - ???
+RMRcvAbort(r) == /\ rmState[r] \= "committed"
+                 /\ [type |-> "Abort"] \in msgs
+                 /\ rmState' = [rmState EXCEPT ![r] = "aborted"]
+                 /\ UNCHANGED << tmState, tmPrepared, msgs >>
+                
+RMRcvCommit(r) == /\ [type |-> "Commit"] \in msgs  
+                  /\ (\/ DoNothing                
+                      \/ RMDoCommit(r))
+                  /\ UNCHANGED << tmState, tmPrepared, msgs >> 
+                
+RMDoCommit(r) == /\ rmState' = [rmState EXCEPT ![r] = "committed"]
+                 /\ (\/ DoNothing
+                     \/ RMConfirmCommit(r) )
+
+\* In fact  - TM got message rm[i] committed
+RMConfirmCommit(r) == /\ msgs' = msgs \union {[type |-> "Commit", rm |-> r]}
+
+\* Increase generation WHEN all answered AND difference is 1
+TMRcvCommitConfirmation(r) == /\ tmGeneration = rmGenerations[r] + 1
+                              /\ rmGenerations[r]' = tmGeneration 
+
+\*-=-=-=-=-=-=-=-=-=-=-
+
+TPNext == \/ TMSendCommit \/ TMSendAbort
+          \/ \E r \in RM :
+            RMRcvPrep(r) \/ RMConfirmPrepare(r) \/ RMRcvAbort(r) 
+            \/ RMRcvCommit(r) \/ RMDoCommit(r) \/ TMRcvCommitConfirmation(r)         
+
+
+\* ---------------
+\* INSTANCE TCommit
+
+\* Invariants
+TCTypeOK == rmState \in [RM -> {"working", "prepared", "pre-committed", "committed", "aborted"}]
+
+TCConsistent == \A r1, r2 \in RM : ~ /\ rmState[r1] = "aborted"
+                                     /\ rmState[r2] = "committed"
```

Comment:
This consistence condition is not correct for this protocol: we need to count only RMs with the latest generation